tools/semgrep/ci/fmt-printf.yml

```diff
@@ -0,0 +1,11 @@
+rules:
+  - id: fmt.Printf
+    languages: [go]
+    message: fmt.Printf is forbidden outside of cmd and test files
+    pattern: fmt.Printf
+    severity: ERROR
+    paths:
+      exclude:
+        - "*_test*.go"
```

Comment:
Is the final asterisk necessary?

Reply:
Oh, probably not. I was just being extra cautious about catching test files. I'll test and remove it.

Reply:
Turns out, it is necessary. There are a few extra test files:

```
  builtin/credential/aws/pkcs7/verify_test_dsa.go
     tools.semgrep.ci.fmt.Printf
        fmt.Printf/Println is forbidden outside of cmd and test files

        111┆ fmt.Printf("%s\n", pemSignature)


  command/server/config_test_helpers.go
     tools.semgrep.ci.fmt.Printf
        fmt.Printf/Println is forbidden outside of cmd and test files

        323┆ fmt.Printf("\n config.Entropy: %#v", config.Entropy)
```

Reply:
I don't think tests in those files are actually being run though. For example, the first test at the top of `verify_test_dsa.go` is called `TestVerifyEC2`.

![CleanShot 2022-05-18 at 15 13 38](https://user-images.githubusercontent.com/947/169164462-e1c2e4d6-7a03-4da9-8158-c7a8b8c44846.png)

Reply:
Possibly, but I think that is out of scope for this PR. Not sure why those tests are effectively disabled. Maybe they are only run manually?

Reply:
I'm sorry, I've done a poor job of explaining my point. What I'm trying to say is that I think `"*_test.go"` is adequate for the exclude pattern, because `go test` ignores files that don't end in `_test.go`. So `verify_test_dsa.go` is not considered a test file by the Go test runner, as my screenshot shows. My concern with `*_test*.go` is having it exclude files that we do want checked, because they're not actual Go test files, e.g. `litmus_test_client.go`, etc.

Reply:
Okay. I'll go ahead and clean up those `fmt.Print*` statements from those files as well then.